writers/batchwriter/batchwriter.go

```diff
@@ -298,7 +298,7 @@ func (w *BatchWriter) Write(ctx context.Context, msgs <-chan message.WriteMessag
 			w.deleteRecordMessages = append(w.deleteRecordMessages, m)
 			l := int64(len(w.deleteRecordMessages))
 			w.deleteRecordLock.Unlock()
-			if w.batchSize > 0 && l > w.batchSize {
+			if w.batchSize > 0 && l >= w.batchSize {
```

Comment:
```suggestion
			limit := batch.CappedAt(0, w.batchSize)
			limit.AddRows(l)
			if limit.ReachedLimit() {
```

I think this is more aligned with the rest of the codebase e.g. https://github.com/cloudquery/plugin-sdk/blob/c82d006af1b3fbc52f6324486df2bfd7efb927dd/writers/streamingbatchwriter/streamingbatchwriter.go#L406

same comment for other places where we check the limit

Reply:
Done in [05b4212](https://github.com/cloudquery/plugin-sdk/pull/2153/commits/05b421261b2d926c13e61cfe9b70e97e8a70b6a5). Tested and it works correctly